tests/clients/test_ngsi_v2_cb.py

```diff
@@ -229,6 +231,147 @@ def test_entity_operations(self):
             client.update_entity(entity=res_entity)
             self.assertEqual(client.get_entity(entity_id=self.entity.id),
                              res_entity)
+            # delete attribute
+            res_entity.delete_attributes(attrs={'pressure': ContextAttribute(
+                type='Number', value=1050)})
+            client.post_entity(entity=res_entity, update=True)
+            self.assertEqual(client.get_entity(entity_id=self.entity.id),
+                             res_entity)
+
+    @clean_test(fiware_service=settings.FIWARE_SERVICE,
+                fiware_servicepath=settings.FIWARE_SERVICEPATH,
+                cb_url=settings.CB_URL)
+    def test_entity_update(self):
+        """
+        Test different ways (post, update, override, patch) to update entity
+        Both with the update scenario
+        1) append attribute
+        2) update existing attribute value
+        1) delete attribute
+        
+        Returns:
+
+        """
+        with ContextBrokerClient(
+                url=settings.CB_URL,
+                fiware_header=self.fiware_header) as client:
+            entity_init = self.entity.model_copy(deep=True)
+            attr_init = entity_init.get_attribute("temperature")
+            attr_init.metadata = {
+                    "metadata_init": {
+                        "type": "Text",
+                        "value": "something"}
+                }
+            attr_append = NamedContextAttribute(**{
+                "name": 'pressure',
+                "type": 'Number',
+                "value": 1050})
+            entity_init.update_attribute(attrs=[attr_init])
+
+            # Post
+            if "post":
+                client.post_entity(entity=entity_init, update=True)
+                entity_post = entity_init.model_copy(deep=True)
+                # 1) append attribute
+                entity_post.add_attributes(attrs=[attr_append])
+                client.post_entity(entity=entity_post, patch=True)
+                self.assertEqual(client.get_entity(entity_id=entity_post.id),
+                                 entity_post)
+                # 2) update existing attribute value
+                attr_append = NamedContextAttribute(**{
+                    "name": 'pressure',
+                    "type": 'Number',
+                    "value": 2050})
+                entity_post.update_attribute(attrs=[attr_append])
+                client.post_entity(entity=entity_post, patch=True)
+                self.assertEqual(client.get_entity(entity_id=entity_post.id),
+                                 entity_post)
+                # 3) delete attribute
+                entity_post.delete_attributes(attrs=[attr_append])
+                client.post_entity(entity=entity_post, update=True)
+                self.assertEqual(client.get_entity(entity_id=entity_post.id),
+                                 entity_post)
+
```

Comment:
@djs0109 After each step (so after `post`, `update_entity `and `override_entity`) I would clear the OCB and reset the variable `attr_append`:

```
clear_all(fiware_header=self.fiware_header, cb_url=settings.CB_URL)
attr_append = NamedContextAttribute(**{
    "name": 'pressure',
    "type": 'Number',
    "value": 1050})
```

Reply:
Good call. I add the `clear_all`, and use `attr_append_update` for updating attribute, so that the `attr_append` will not be touched during the whole test.